src/st/st-theme-node-drawing.c

```diff
@@ -590,18 +590,44 @@ create_cairo_pattern_of_background_gradient (StThemeNode *node,
     pattern = cairo_pattern_create_linear (0, 0, width, 0);
   else
     {
-      gdouble cx, cy;
+      gdouble cx, cy, radius;
 
-      cx = width / 2.;
-      cy = height / 2.;
-      pattern = cairo_pattern_create_radial (cx, cy, 0, cx, cy, MIN (cx, cy));
+      if (node->background_gradient_position_set)
+        {
+          cx = node->background_gradient_position_x;
+          cy = node->background_gradient_position_y;
+        }
+      else
+        {
+          cx = width / 2.;
+          cy = height / 2.;
+        }
+
+      if (node->background_gradient_radius != -1)
+        radius = node->background_gradient_radius;
+      else
+        radius = MIN (width / 2., height / 2.);
```

Comment:
Shouldn't this be `MIN (cx, cy)`  ?

Reply:
No. If the `background_gradient_radius == -1`, the code strictly falls back to the previous behavior.

I did this solely to reduce the diff from the theme, since otherwise the theme itself would need to be adapted, and having bigger changes on the edge of a rebase feels like a bad decision.

Reply:
Ok